d/aws_route53_resolver_rules: add new name_regex filter argument (#24582)
hashicorp/terraform-provider-aws: website/docs/d/route53_resolver_rules.html.markdown

## Example Usage

> ### Retrieving the default resolver rule

Reviewer: Thanks for adding the headings

}
```

> ### Retrieving forward rule(s) shared with me

Reviewer: For stylistic reasons, we prefer `rules`

```suggestion
### Retrieving forward rules shared with me
```

```terraform
data "aws_route53_resolver_rules" "example" {
  rule_type    = "FORWARD"
  share_status = "SHARED_WITH_ME"
}
```

> ### Retrieving rule(s) by name regex

Reviewer: ```suggestion
### Retrieving rules by name regex
```

Resolver rules whose name contains `abc`.

```terraform
data "aws_route53_resolver_rules" "example" {
  name_regex = ".*abc.*"
}
```

## Argument Reference

The arguments of this data source act as filters for querying the available resolver rules in the current region.

> * `name_regex` - (Optional) A regex string to apply to the resolver rules list returned by AWS. This allows more advanced filtering not supported from the AWS API.
>   This filtering is done locally on what AWS returns, and could have a performance impact if the result is large. It is recommended to combine this with other
>   options to narrow down the list AWS returns.

Reviewer: Some suggested changes to the argument description

```suggestion
* `name_regex` - (Optional) A regex string to filter resolver rule names.
  The filtering is done locally, so could have a performance impact if the result is large.
  This argument should be used along with other arguments to limit the number of results returned.
```
